### ISSUE TITLE
Change the default value of `editors/2d/constrain_editor_view` to false

### ISSUE THREAD

**Issue by Ark2000** [godotengine/godot-proposals#6894]
### Describe the project you are working on

A simple platformer game in Godot 4.0.2

### Describe the problem or limitation you are having in your project

https://github.com/godotengine/godot-proposals/assets/21126555/e2837d3d-a9f3-4fbd-ad94-c32250528e18

Currently when zooming in and out in the 2D editor, the view becomes unstable and jumpy, providing a poor user experience. This makes precise editing and placement of nodes difficult and frustrating.

### Describe the feature / enhancement and how it helps to overcome the problem or limitation

The proposal is to disable `constrain_editor_view` by default in the 2D editor. This will make the view focus stay the same when zooming, providing a smooth zooming experience. This enhances the usability of the 2D editor and improves the overall user experience of Godot.

### Describe how your proposal will work, with code, pseudo-code, mock-ups, and/or diagrams

![image](https://github.com/godotengine/godot-proposals/assets/21126555/ab7a3508-608f-4b44-a9bf-e8bf2790d720)

Simply set `constrain_editor_view` to false by default in the 2D editor. This can be implemented by modifying the following line:

```cpp
_initial_set("editors/2d/constrain_editor_view", false);
```

### If this enhancement will not be used often, can it be worked around with a few lines of script?

While this can be workaround by users manually disabling `constrain_editor_view`, many users may not realize this option exists and will continue suffering from the poor experience. As an editor tool, providing a smooth user experience should be the priority.

### Is there a reason why this should be core and not an add-on in the asset library?

Improving the editor user experience is critical to Godot's success as an open source game engine. An unfavorable editor experience can negatively impact users' impression of Godot and hinder its adoption. As a simple fix that significantly enhances usability, this proposal should be implemented as a core improvement to the engine.

In summary, this proposal aims to greatly improve the 2D editing experience in Godot by providing smooth view zooming. As an easy implementation with big impact, it should be adopted as a core improvement to enhance Godot's editor usability and overall user experience. Please let me know if you have any other questions or require any clarification.

**Comment by YuriSizov** [comment]
https://github.com/godotengine/godot/pull/47628 completely removed the constraint, solving the underlying problem.
